### DISCUSSION TITLE
Is it secure enough to be used for projects ?

### DISCUSSION THREAD

**Discussion by rthangam** [community/community#7737] [Codespaces]
I am curious to know if this tool can be used for coding projects which are proprietary, because i see editor like VS Code pass the content to the engine to get sample recommendations, which can be a problematic one for proprietary projects since the engine will have access to the code contents.

**Comment by metaskills** [comment]
I think so, we use  VS Code heavily and Codespaces at Custom Ink for our internal projects.

**Reply by rthangam** [reply]
Thanks @metaskills

**Comment by asciimike** [comment]
The Codespaces team [publishes security best practices](https://docs.github.com/en/codespaces/codespaces-reference/security-in-codespaces) that aims to address this question. Please take a look and let me know if it answers your questions, or if there's additional content we should add.

**Reply by rthangam** [reply]
Thanks @asciimike  i will check it out

**Reply by numbsafari** [reply]
This is very helpful. Thanks!
